pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,417 @@
+    time.sleep(1.0)  # Give time to output on
+
+    abs_output_voltage_error = 0.02 # kV
+
+    hpp120256.wait_for_output_voltage_reached(abs_output_voltage_error, 1.0, 40.0)
+
+    # Here voltage HV output should be at 0.0 kV
+
+    print("Setting the output voltage to 1.0kV...")
+    hpp120256.voltage_setpoint = 1.0  # kV
+
+    # Now HV output should be rising to reach the 1.0kV at 50.0 V/s
+
+    hpp120256.wait_for_output_voltage_reached(abs_output_voltage_error, 1.0, 40.0)
+
+    # Here voltage HV output should be at 1.0 kV
+
+    hpp120256.shutdown(100.0)
+
+    hpp120256.wait_for_output_voltage_reached(abs_output_voltage_error, 1.0, 60.0)
+
+    # Here voltage HV output should be at 0.0 kV
+
+    inst.enable_output = False
+
+    # Now the HV voltage source is in safe state
+
+    """
+
+    VOLTAGE_RANGE = [0.0, 12.0]  # kVolts
+    CURRENT_RANGE = [0.0, 25.0]  # mAmps
+    VOLTAGE_RAMP_RANGE = [10, 3000]  # V/s
+    COMMAND_DELAY = 0.2  # s
+
+    response_encoding = "iso-8859-2"
+    f_numbers_regex_pattern = r'([+-]?([\d]*\.)?[\d]+)'
+    regex = re.compile(f_numbers_regex_pattern)
+
+    def __init__(self,
+                 adapter,
+                 query_delay=0.1,
+                 write_delay=0.4,
+                 timeout=5000,
+                 **kwargs):
+
+        super().__init__(
+            adapter,
+            "Euro Test High Voltage DC Source model HPP-120-256",
+            write_termination="\n",
+            read_termination="",
+            send_end=True,
+            includeSCPI=False,
+            timeout=timeout,
+            **kwargs
+        )
+
+        self.write_delay = write_delay
+        self.query_delay = query_delay
+
+    # ####################################
+    # # EuroTest-Command set. Non SCPI commands.
+    # ####################################
+
+    voltage_setpoint = Instrument.control(
+        "STATUS,U", "U,%.3fkV",
+        """ A floating point property that represents the output voltage
+        setting (in kV) of the HV Source. This property can be set.""",
+        # getter device response: "U, RANGE=3.000kV, VALUE=2.458kV"
+        validator=strict_range,
+        values=VOLTAGE_RANGE,
+        get_process=lambda r:
+        float(EurotestHPP120256.regex.search(r[2].strip()).groups()[0])
+    )
+
+    current_limit = Instrument.control(
+        "STATUS,I", "I,%.3fmA",
+        """ A floating point property that represents the output current limit setting (in mA)
+        of the HV Source. This property can be set.""",
+        # When this property acts as get, the instrument will return a string like this:
+        # "I, RANGE=5000mA, VALUE=1739mA", then current_limit will return 1739.0,
+        # hence the convenience of the get_process.
+        validator=strict_range,
+        values=CURRENT_RANGE,
+        get_process=lambda r:
+        float(EurotestHPP120256.regex.search(r[2].strip()).groups()[0])
+    )
+
+    voltage_ramp = Instrument.control(
```

Comment:
```suggestion
    voltage_ramp_rate = Instrument.control(
```
Better name? Also needs to be changed below, then.

Reply:
I think voltage_ramp denotes itself a rate (volts/secs)

Reply:
@sansanda Please don't press "resolve" if the discussion is not resolved, especially if you expect feedback! (that button hides the discussion from view by default)

In general, if it's not a very clear change, like committing a suggested change, let the one who opened a discussion resolve it, that way both parties can check that it is indeed resolved.

Reply:
> I think voltage_ramp denotes itself a rate (volts/secs). I prefer to leave it as is, don't you think?

Well, that's of course a matter of taste up to some point. I'll point out that even your docstring refers to "ramp speed" (more accurately a rate, not a speed, but that's beside the main point), so I'd think that `voltage_ramp_rate` would be clearer. :-)

Reply:
However, it's a minor thing, I'll leave it up to you.